Add custom MLIR-TensorRT usage instruction to Tripy CONTRIBUTING.md (#49)
NVIDIA/TensorRT-Incubator: tripy/CONTRIBUTING.md

# Contributing To Tripy

Thank you for considering contributing to Tripy!

## Table of Contents
- [Setting Up](#setting-up)
- [Making Changes](#making-changes)
  - [Before You Start: Install pre-commit](#before-you-start-install-pre-commit)
  - [Getting Up To Speed](#getting-up-to-speed)
  - [Making Commits](#making-commits)
- [Tests](#tests)
- [Documentation](#documentation)

> - [Use Custom MLIR-TensorRT with Tripy](#use-custom-mlir-tensorrt-with-tripy)

Reviewer: GitHub autogenerates a table of contents. I think we can probably drop this?

If you're intersted in adding a new operator to Tripy, refer to [this guide](./docs/post0_developer_guides/how-to-add-new-ops.md).

### Making Commits

1. Tripy follows [fork based developement](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo).
    - Fork repo from GitHub
    - Push changes to your branches on fork instead of the main repo
    - Create a PR to merge changes from fork to the main repo

2. Managing PRs

>     - Label your PR correctly (e.g., use `tripy` for `tripy` repository).

Reviewer: ```suggestion
    - Label your PR correctly (e.g., use `tripy` for changes to `tripy`).
```

>		this project or the open source license(s) involved.

### Use custom MLIR-TensorRT with Tripy
Tripy depends on [MLIR-TensorRT](../mlir-tensorrt/README.md) for compilation and execution.

> Tripy container currently builds with [mlir-tensorrt-v0.1.29](https://github.com/NVIDIA/TensorRT-Incubator/releases/tag/mlir-tensorrt-v0.1.29), but you may
> choose to test Tripy with custom version of MLIR-TensorRT.

Reviewer: ```suggestion
The Tripy container currently builds with [mlir-tensorrt-v0.1.29](https://github.com/NVIDIA/TensorRT-Incubator/releases/tag/mlir-tensorrt-v0.1.29), but you may
choose to test Tripy with a custom version of MLIR-TensorRT.
```

1. [Build custom MLIR-TensorRT](#contributing-to-mlir-tensorrt)

> 2. Launch the container; from the [`tripy` root directory](.) with mlir-tensorrt repository mapped for accessing wheels files, run:

Reviewer: ```suggestion
2. Launch the container with mlir-tensorrt repository mapped for accessing wheels files; from the [`tripy` root directory](.), run:
```

    ```bash
    docker run --gpus all -it -v $(pwd):/tripy/ -v $(pwd)/../mlir-tensorrt:/mlir-tensorrt  --rm tripy:latest
    ```

3. Install MLIR-TensorRT wheels
    MLIR-TensorRT builds with a specific version of TensorRT, ensure it matches the version required by Tripy.

>     This requirement can be relaxed in future.

Author: @parthchadha Looks like we build MLIR-TensorRT container using [mlir-tensorrt-compiler==0.1.29+cuda12.trt102](https://github.com/NVIDIA/TensorRT-Incubator/blob/main/tripy/pyproject.toml#L10) and install `TensorRT-10.1.0.27` in the [Dockerfile](https://github.com/NVIDIA/TensorRT-Incubator/blob/main/tripy/Dockerfile#L28).

Reviewer 2: That's intentional. We build `mlir-tensorrt` with 10.2 and can link with 10.0+ (linking with 10.1 to keep it compatible with TRT-LLM).
